File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        left=0
        right= len(arr)
        while left < right:
            m=int((right+left)/2)
            if arr[m]-m-1<k:
                left=m+1
            else:
                right=m
        return left+k

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int left=0;
        int right=arr.length;
        int m;
        while(left < right){
            m=(left+right)/2;
            if(arr[m]-m-1<k){left=m+1;}
            else{right=m;}
        }
        return left+k;
    }
}